### PR TITLE
zoe: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **zoe/***

zoe is renamed from teemo.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/zoe/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+sources:
+  "3.0":
+    url: "https://github.com/winsoft666/zoe/archive/refs/tags/v3.0.tar.gz"
+    sha256: "718c6de993bea2a4b1866070a8685e77cfa7929bbd9e22abb29e54c67a687a5e"
+
+patches:
+  "3.0":
```
Comment (RubenRBS) [inline recipes/zoe/all/conandata.yml:7]: Do you plan to submit the other two patches upstream?

File: recipes/zoe/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+required_conan_version = ">=1.53.0"
+
+class ZoeConan(ConanFile):
+    name = "zoe"
+    description = "C++ File Download Library."
+    license = "GPL-3.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/winsoft666/zoe"
+    topics = ("downloader", "libcurl", "speed-limit", "ftp", "http")
+    settings = "os", "arch", "compiler", "build_type"
```
Comment (RubenRBS) [inline recipes/zoe/all/conanfile.py:19]: Missing a `package_type` attribute

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot! A few other questions in this one too :)

File: recipes/zoe/all/conandata.yml (hunk 1 above)
Comment (toge) [reply recipes/zoe/all/conandata.yml:7]: Thanks!
I am going to create PR for two.

File: recipes/zoe/all/conanfile.py (hunk 2 above)
Comment (toge) [reply recipes/zoe/all/conanfile.py:20]: fixed

**Review by jwillikers — CHANGES_REQUESTED** [review]

**Review by jwillikers — APPROVED** [review]

File: recipes/zoe/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("libcurl/8.2.1")
```
Comment (uilianries) [inline recipes/zoe/all/conanfile.py:49]:
```suggestion
        self.requires("libcurl/[>=8.21 <9]")
```

We are accepting libcurl with version ranges already 😉  https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges

File: recipes/zoe/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("libcurl/[>=8.21 <9]")
```
Comment (RubenRBS) [inline recipes/zoe/all/conanfile.py:49]: @toge what version were you planning on using here? The latest available libcurl version is `8.6` :)

**Review by RubenRBS — CHANGES_REQUESTED** [review]
Thanks!

**Review by jwillikers — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]
Thanks!

**Pull request merged**